diff --git a/src/hydra-queue-runner/queue-monitor.cc b/src/hydra-queue-runner/queue-monitor.cc
index 862925b2..550be546 100644
--- a/src/hydra-queue-runner/queue-monitor.cc
+++ b/src/hydra-queue-runner/queue-monitor.cc
@@ -90,10 +90,17 @@ bool State::getQueuedBuilds(Connection & conn,
     {
         pqxx::work txn(conn);
 
-        auto res = txn.parameterized
+        pqxx::result res;
+        if (lastBuildId == 0) {
+          res = txn.exec
             ("select id, project, jobset, job, drvPath, maxsilent, timeout, timestamp, globalPriority, priority from Builds "
-             "where id > $1 and finished = 0 order by globalPriority desc, id")
+             "where finished = 0 order by globalPriority desc, id");
+        } else {
+          res = txn.parameterized
+            ("select id, project, jobset, job, drvPath, maxsilent, timeout, timestamp, globalPriority, priority from Builds "
+             "where id > $1 order by globalPriority desc, id")
             (lastBuildId).exec();
+        }
 
         for (auto const & row : res) {
             auto builds_(builds.lock());
diff --git a/src/script/hydra-eval-jobset b/src/script/hydra-eval-jobset
index 9d375c1b..711a1df4 100755
--- a/src/script/hydra-eval-jobset
+++ b/src/script/hydra-eval-jobset
@@ -746,7 +746,7 @@ sub checkJobsetWrapped {
             # Wake up hydra-queue-runner.
             my $lowestId;
             while (my ($id, $x) = each %buildMap) {
-                $lowestId = $id if $x->{new} && (!defined $lowestId || $id < $lowestId);
+                $lowestId = $id if (!defined $lowestId || $id < $lowestId);
             }
             $notifyAdded->execute($lowestId) if defined $lowestId;
 

